feat/feature.go

```diff
@@ -140,33 +165,42 @@ func PositionWithin(f, ref Feature, position int) (pos int, ok bool) {
 	panic("feat: feature chain too long")
 }
 
-// BaseOrientationOf returns the orientation of the given feature relative to
-// the deepest orientable location and the reference feature, which may be
-// the feature itself if it is not an Orienter or has a nil Location.
-// The returned orientation will always be Forward or Reverse.
+// BaseOrientationOf returns the orientation of f relative to the shallowest
```

Comment:
I don't think this description is accurate. It does not distinguish between a non-orientable and a nil. I does not capture the following.

... a reference feature which will be the shallowest non orientable feature location **if it is not nil or its parent feature if it is nil**.